src/Logger.php

```diff
@@ -82,22 +73,21 @@ final class Logger implements LoggerInterface
      */
     private int $flushInterval = 1000;
 
-    /**
-     * @var int How much call stack information (file name and line number) should be logged for each log message.
-     *
-     * If it is greater than 0, at most that number of call stacks will be logged.
-     * Note that only application call stacks are counted.
-     */
-    private int $traceLevel = 0;
+    private ContextEnricherInterface $contextEnricher;
 
     /**
      * Initializes the logger by registering {@see Logger::flush()} as a shutdown function.
      *
      * @param Target[] $targets The log targets.
+     * @param ContextEnricherInterface|null $contextEnricher The context enricher. If null, {@see ContextEnricher} with
+     * default parameters will be used.
      */
-    public function __construct(array $targets = [])
-    {
+    public function __construct(
+        array $targets = [],
+        ?ContextEnricherInterface $contextEnricher = null,
```

Comment:
Should we provide a composite out of the box to be able to combine multiple context providers?

Reply:
Added `CompositeContextEnricher`.